eclipse-che/che-devfile-registry
# Add Bitbucket and Bitbucket Server support

### What does this PR do?
Add Bitbucket and Bitbucket Server support

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-4323

### How to test this PR?

1. Deploy bitbucket server: https://github.com/che-incubator/git-srv/tree/main/bitbucket.
2. Create a repository with a devfile in the bitbucket server instance.
3. Go to projects directory: '/tools/devworkspace-generator' directory
4. execute `yarn`
5. execute 
```bash
node lib/entrypoint.js \
        --plugin-registry-url:https://che-plugin-registry-main.surge.sh/v3/ \
        --editor-entry:che-incubator/che-code/latest \
        --output-file:<output directory> \
        --injectDefaultComponent:true \
        --defaultComponentImage:registry.access.redhat.com/ubi8/openjdk-11:latest \
        --devfile-url:<url of the bitbucket server repo>
```
See: `yaml` file is generated and contains the devfile from the repository.
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
